src/components/Dropdowns/AccessibleDropdown.tsx

```diff
@@ -0,0 +1,242 @@
+const MainButtonContents = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+
+  // Button text label
+  & div {
+    width: 100%;
+    text-align: left;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
+  // Dropdown arrow
+  & svg {
+    width: 14px;
+    height: 14px;
+  }
+`;
+
+/** Wrapper around the dropdown content. */
+const DropdownContentWrapper = styled.div`
+  padding: 4px;
+`;
+
+/**
+ * A keyboard-accessible wrapper around the Antd `Dropdown` component, with support for button styling, labels, and tooltips.
+ */
+export default function AccessibleDropdown(inputProps: AccessibleDropdownProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<AccessibleDropdownProps>;
+
+  //// Handle clicking on the dropdowns ///////////////////////////////////
+
+  // Support tab navigation by forcing the dropdown to stay open when clicked.
+  const [forceOpen, setForceOpen] = useState(false);
+  const componentContainerRef = useRef<HTMLDivElement>(null);
+
+  // If open, close the dropdown when focus is lost.
+  // Note that the focus out event will fire even if the newly focused element is also
+  // inside the component, so we need to check if the new target is also a child element.
+  useEffect(() => {
+    if (!forceOpen) {
+      return;
+    }
+    const doesContainTarget = (target: EventTarget | null): boolean => {
+      return (
+        (target instanceof Element &&
+          componentContainerRef.current &&
+          componentContainerRef.current.contains(target)) ||
+        false
+      );
+    };
+    const handleFocusLoss = (event: FocusEvent): void => {
+      if (!doesContainTarget(event.relatedTarget)) {
+        setForceOpen(false);
+      }
+    };
+
+    componentContainerRef.current?.addEventListener("focusout", handleFocusLoss);
+    return () => {
+      componentContainerRef.current?.removeEventListener("focusout", handleFocusLoss);
+    };
+  }, [forceOpen]);
+
+  //// Handle rendering of buttons and dropdown contents ///////////////////
+  const defaultRenderButton = (props: AccessibleDropdownProps, isOpen: boolean, onClick: () => void): ReactElement => {
+    return (
+      <MainButton
+        disabled={props.disabled}
+        style={props.buttonStyle}
+        $type={props.buttonType}
+        type={props.buttonType === "outlined" ? "default" : props.buttonType}
+        $open={isOpen}
+        // Open the dropdown when clicked for accessibility
+        onClick={onClick}
+      >
+        <MainButtonContents>
+          <div>
+            {props.buttonText}
+            <VisuallyHidden>(click to open menu)</VisuallyHidden>
+          </div>
+          <DropdownSVG />
+        </MainButtonContents>
+      </MainButton>
+    );
+  };
+  const renderButton = props.renderButton || defaultRenderButton;
```

Comment:
Given that none of the current implementations are providing a `renderButton` prop could you reduce the complexity in this file by just having the default, or do you anticipate instances that will need custom button rendering functions?

Reply:
Hmm, I think because both of the optional props I've included don't increase the complexity by a *huge* amount, I'm willing to leave them as-is. It was useful to think about making the construction steps of the dropdown more modular.

Admittedly, I didn't use either of the optional props in #244, which is where I expected to need them, though.

Reply:
Continuing from my comment above, this could become a separate component outside the body of this one, with a type that it shares with the type of that field in `AccessibleDropdownProps`. 100% optional suggestion, just a thought.